### DISCUSSION TITLE
Can I use `timeout()` on `once()`?

### DISCUSSION THREAD

**Discussion by Norlandz** [socketio/socket.io#4859] [Q&A] [answered]
I know this::
> `socket.timeout(5000).emit("my-event", (err) => {`
> https://socket.io/docs/v4/client-api/#sockettimeoutvalue

But is this possible::
`socket.timeout(5000).once("offerSent", (webrtcConnectionEvent_jsobj: WebrtcConnectionEvent) => {`
- timeout if no message received in this event listener.

If not, do I have to use something like a `setTimeout()`?
```js
socket.once(eventSessionId, listener);
setTimeout(() => socket.off(eventSessionId, listener), eventSessionLifetimelength);
```

**Comment by darrachequesne** [comment, marked as answer]
Hi!

No it's not possible. The solution you provided with the `setTimeout()` should work well :+1:
